xwiki-platform-core/xwiki-platform-refactoring/xwiki-platform-refactoring-api/src/main/java/org/xwiki/refactoring/RefactoringConfiguration.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.refactoring;
+
+import org.xwiki.component.annotation.Role;
+import org.xwiki.stability.Unstable;
+
+/**
+ * Provides configuration for the refactoring module.
+ *
+ * @version $Id$
+ * @since 12.8RC1
+ */
+@Role
+@Unstable
+public interface RefactoringConfiguration
+{
+    /**
+     *
+     * @return {@code true} if the configuration allows advanced users to skip the recyclebin when deleting documents.
+     */
+    boolean canSkipRecyclebin();
```

Comment:
This is an API so very important to get right. I don't understand this API: why is it not `isSentToRecycleBin()` or another positive terminology (`isRecycled()`, etc) signifying that deleted documents (is it only documents or also attachments? - if only docs then we might need to mention Documents in the API) are sent to the recycle bin or not. Is the refactoring module only for docs?

Reply:
I think the behaviour when the user choose to permanently remove a document should be the same as when a wiki does not have a recycle bin.
I am not entirely sure what is the behaviour in this case but I'll investigate in this direction.

Reply:
And to explain this API further, this method returns true if the current user has the right to decide by himself if he wants to send the document to the recycle bin or the remove it permanently. I'll update the javadoc it make it more explicit.